### PR TITLE
Feat/elastic search

### PR DESCRIPTION
**Pull request by annelhote**
Add indexation in ES for structures' names

### CODE REVIEW

File: src/api/commons/middlewares/crud-nested.middlewares.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { NotFoundError, ServerError } from '../http-errors';
 
 const read = (repository, useQuery) => async (req, res, next) => {
-  const { id, resourceId } = req.params;
+  const { id, resourceId } = req?.params || {};
```
Comment (folland87) [inline src/api/commons/middlewares/crud-nested.middlewares.js:4]: is there any way req.params isn't set when we call controllers.<function> ?
If yes, shouldn't we let it fail ?

File: src/api/commons/queries/names.elastic.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import metas from './metas.query';
+
+export default [
+  ...metas,
+  {
+    $project: {
+      _id: 0,
+      officialName: { $ifNull: ['$usualName', null] },
```
Comment (folland87) [inline src/api/commons/queries/names.elastic.js:8]: I think there is no need for ifNull in that case. If it does not exists you don't want it to be indexed anyway.
Usually i use it to return the value in the api response even if it's missing.

File: src/api/commons/queries/names.query.js [hunk 3]
```diff
@@ -7,13 +7,14 @@ export default [
       _id: 0,
       id: 1,
       resourceId: 1,
-      officialName: 1,
+      officialName: { $ifNull: ['$usualName', null] },
```
Comment (folland87) [inline src/api/commons/queries/names.query.js:10]: officialName is required at creation and non nullable at patch time so it should always be there. If not, response will throw error 500 which is what I would expect.

**Review by folland87 — COMMENTED** [review]
One case is missing I think. When a structure is created with names at /structures, it won't be indexed untill a name is changed.

File: src/api/commons/middlewares/crud-nested.middlewares.js (hunk 1 above)
Comment (annelhote) [reply src/api/commons/middlewares/crud-nested.middlewares.js:4]: Fixed

File: src/api/commons/queries/names.elastic.js (hunk 2 above)
Comment (annelhote) [reply src/api/commons/queries/names.elastic.js:8]: Fixed

File: src/api/commons/queries/names.query.js (hunk 3 above)
Comment (annelhote) [reply src/api/commons/queries/names.query.js:10]: Fixed

**Review by folland87 — APPROVED** [review]

**Pull request merged**